### PR TITLE
Migrate to bootstrap 4

### LINKED ISSUE #1194: Migrate to bootstrap 4
- [ ] Drop `.visible*` classes, they no longer exist; replace with use of appropriate `.d-*` classes or media breakpoints.
- [ ] Bump up grid tiers, but not for buttons sizes (`btn-sm` stays as is).
- ...

### PR DESCRIPTION
**Pull request by chalin**
Closes #1194

Staged at https://ng2-io.firebaseapp.com

cc @filiph @fertolg

### CODE REVIEW

**Comment by Sfshaza** [comment]
Wow! What a lot of work, @chalin. I scanned the file changes and played with the staged site. It looks great!

**Comment by Sfshaza** [comment]
LGTM from me, but waiting for @filiph to weigh in.

File: src/_includes/head.html [hunk 1]
```diff
@@ -27,7 +27,7 @@
   {% assign og_image_path = page.image | default: layout.image | default: '/images/flutter-logo-sharing.png' %}
   <meta property="og:image" content="{{og_image_path | absolute_url}}">
 
-  <link href="https://fonts.googleapis.com/icon?family=Material+Icons+Extended" rel="stylesheet">
+  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
```
Comment (filiph) [inline src/_includes/head.html]: Are we sure we don't need the Extended set? I _think_ we don't, but I'd rather err on the side of caution.

File: src/_includes/header.html [hunk 2]
```diff
@@ -1,31 +1,29 @@
...
-              <input type="hidden" name="hl" value="en">
-              <input type="search" name="q" id="q" autocomplete="off" placeholder="Search">
-            </form>
-          </div>
-        </nav>
+    <div class="site-nav-horizontal">
+      {% include main-nav.html class="nav-item" %}
+    </div>
+
+    <div class="site-nav-vertical dropdown">
```
Comment (filiph) [inline src/_includes/header.html]: [nit] The current version shows the drop down on mouse-over, and it also shows `cursor: hand`.

File: src/_layouts/default.html [hunk 3]
```diff
@@ -10,20 +10,20 @@
...
+        <div id="side-nav-container" class="col-md-3">
           {% include sidebar.html %}
         </div>
 
         {% if page.url == '/' %}
         {% comment %} Homepage {% endcomment %}
 
         <!-- Homepage Content Column -->
-        <div class="col-sm-12 main-contents">
+        <div class="col-md-12 main-contents">
```
Comment (filiph) [inline src/_layouts/default.html]: I don't understand why the change from `sm` to `md` here and elsewhere. Should we always use `sm` by default and then only override for bigger screen? (I thought that's the philosophy of bootstrap's responsive design.)

Comment (chalin) [reply src/_layouts/default.html]: Because a [new tier was added to BS 4](https://getbootstrap.com/docs/4.0/migration/#grid-system), so every tier has been bumped up one level:`sm` -> `md` -> `lg` -> `xl`.

Comment (chalin) [reply src/_layouts/default.html]: (Thanks for the feedback. I'm heading out, so I'll respond to you other comments later.)

Comment (chalin) [reply src/_layouts/default.html]: Just to be clear: the media breakpoints of both the current and this new version of the site are the same (e.g., 710px is one breakpoint for the home page). It is just that break point names have changed.

File: src/_includes/head.html (hunk 1 above)
Comment (chalin) [reply src/_includes/head.html]: AFAIK, this is how one might use an icon from the extended collection:

- Reference the font family (`font-family: 'Material Icons Extended';`) directly, or
- Use the CSS class `.material-icons-extended`, or
- `<span class="mdi mdi-file-upload-outline"></span>`, or
- Use a direct link: https://petershaggynoble.github.io/MDI-Sandbox/?icon=file-upload-outline, or
- Use a local image of the icon, but in that case we don't need to load the font.

(If you can think of other ways, let me know.)

The site source contains no such uses.

File: src/_includes/header.html (hunk 2 above)
Comment (chalin) [reply src/_includes/header.html]: The hover behavior is non-standard for menus/dropdowns, and doesn't work on mobile. I'll address the cusor in a followup PR.

Comment (chalin) [reply src/_includes/header.html]: > I'll address the cursor in a followup PR.

Fixed in #1220.

File: src/_includes/head.html (hunk 1 above)
Comment (filiph) [reply src/_includes/head.html]: Ah, thanks for the due diligence on this.

File: src/_layouts/default.html (hunk 3 above)
Comment (filiph) [reply src/_layouts/default.html]: Thanks for clarifying!
